src/coreclr/gc/gc.cpp

```diff
@@ -26685,7 +26685,7 @@ void gc_heap::add_to_hc_history_worker (hc_history* hist, int* current_index, hc
     current_hist->concurrent_p = (bool)settings.concurrent;
     current_hist->bgc_thread_running = (bool)bgc_thread_running;
 
-#if defined(TARGET_AMD64) && defined(TARGET_WINDOWS) && !defined(_DEBUG)
+#if defined(TARGET_AMD64) && defined(TARGET_WINDOWS) && !defined(_DEBUG) && !defined(FEATURE_NATIVEAOT)
     int bgc_thread_os_id = 0;
 
     if (bgc_thread)
```

Comment:
This magic 0x130 constant is going to break in number of different scenarios. For example, mix and match standalone GC scenarios are going to break too.

How long do we expect to keep it?

Reply:
I'm planning to make this into a GCToEEInterface call in main and possibly backport to 9.0 in a servicing release.

Reply:
I think we should do the proper change now.

Reply:
I'm only intending to use this with 9.0 so I'm fine not supporting mix/match standalone GC. other than coreclr actually changing the Thread object, how else can this break?

Reply:
should we also consider instrumenting under a special config?

Reply:
yep, we can put this under a config.

Reply:
ok I can add that, just this piece or the entire instrumentation change?

Reply:
I would only do the config for this. thanks!

Reply:
added a config check for it.

Reply:
LGTM.